### DISCUSSION TITLE
Add :noCollideWorld

### DISCUSSION THREAD

**Discussion by Isemenuk27** [wiremod/wire#2780] [Suggestions]
You can make entity have no collisions with world entity via Lua "constraint.NoCollide( Ent, game.GetWorld(), 0, 0 )", but cant do with E2 in multiplayer as you dont/cant own world entity.

```
e2function void entity:noCollideWorld(entity target)
        if not IsValid(this) then return self:throw("Invalid entity!", false) end
        if not isOwner(self, this) then return self:throw("You are not the owner of the entity!", false) end
        if entity:IsPlayer() then return self:throw("Cannot constrain players!", false) end
        constraint.NoCollide(ent1, ent2, 0, 0)
end
```

**Comment by Isemenuk27** [comment]
```
@name 111
@model models/props_c17/display_cooler01a.mdl
@strict

world():noCollide( entity() )
```

--> sv: Expression 2 (111): Runtime error 'You are not the owner of the entity!' at line 5, char 9

[gm_aknevo_morning 2023-10-5 18-12-36.webm](https://github.com/wiremod/wire/assets/77487254/d1dc3831-f42d-40b9-85ac-7b4872f2e090)

**Reply by Denneisk** [reply]
It works fine if you no-collide with the entity first and then the world, but that seems a bit unintentional.

Edit: May just work because I'm in sandbox singleplayer without any PP though.

**Comment by vurvdev** [comment]
Shouldn't need a separate function for this, just allow making nocollide constraints to the world.

**Comment by vurvdev** [comment]
https://github.com/wiremod/wire/pull/2782
